File: CodeGolf/Temperature/Temperature.py
# Normal version

n,a=int(input()),[int(i)for i in input().split()]
m=(0,999)[n>0]
for i in a:m=(m,i)[abs(m)>abs(i)or(abs(m)==abs(i)and i>m)]
print(m)

# Code golf

s=abs
z=int
w=input
n,a=z(w()),[z(i)for i in w().split()]
m=(0,999)[n>0]
for i in a:m=(m,i)[s(m)>s(i)or(s(m)==s(i)and i>m)]
w(m)

File: CodeGolf/Temperature/Temperature.js
// Readable version

m=Math.abs
z=readline;
n=z();
(t=z().split(" "))!=''?r=t[0]:r=0
for (j of t){
    l=m(r);
    z=m(j);
    if (l>z || l==z && r<j) r=j
}
print(r)

// Code Golf

m=Math.abs;z=readline;n=z();(t=z().split(" "))!=''?r=t[0]:r=0
for(j of t){l=m(r);z=m(j);if(l>z||l==z&&r<j)r=j}print(r)